include/xtensor/xcontainer.hpp

```diff
@@ -658,6 +664,30 @@ namespace xt
         return storage()[i];
     }
 
+    template <class D>
+    inline auto xcontainer<D>::front() -> reference
+    {
+        return *this->begin();
+    }
+
+    template <class D>
+    inline auto xcontainer<D>::front() const -> const_reference
+    {
+        return *this->begin();
+    }
+
+    template <class D>
+    inline auto xcontainer<D>::back() -> reference
+    {
+        return *std::prev(this->end());
+    }
+
+    template <class D>
+    inline auto xcontainer<D>::back() const -> const_reference
+    {
+        return *std::prev(this->end());
+    }
```

Comment:
This should actually be moved into `xaccessible` / `xconst_accessible`. And then all you have to do is making them visible in inheriting classes with using declarations.

Reply:
Yeah so I had that, but:
```

Reply:
My bad, the implementation should be
```cpp
return *std::prev(derived_cast().end());
```